compose/config/interpolation.py

```diff
@@ -41,7 +81,9 @@ def interpolate_value(name, config_key, value, section, mapping):
 
 def recursive_interpolate(obj, mapping):
     if isinstance(obj, six.string_types):
-        return interpolate(obj, mapping)
+        if '$' in obj:
+            return interpolate(obj, mapping)
+        return obj
```

Comment:
Was this necessary, or is it just a shortcut for the case where there are no variables?